tests/components/RangeInput.test.tsx

```diff
@@ -0,0 +1,296 @@
+    const minTextbox = screen.getAllByRole('textbox')[0];
+    userEvent.type(minTextbox, '10');
+    await waitFor(() => {
+      expect(minTextbox).toHaveValue('10');
+    });
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(minTextbox).toHaveValue('');
+  });
+});
+
+describe('Renders correctly for max input', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+  it('renders input value, clear, and apply option when inputing max value', async () => {
+    render(
+      <FilterGroupContext.Provider value={filterGroupContextValue}>
+        <FiltersContext.Provider value={filterContextValue}>
+          <RangeInput />
+        </FiltersContext.Provider>
+      </FilterGroupContext.Provider>);
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    expect(screen.getByText('Apply')).toBeDefined();
+  });
+
+  it('record proper values in state when applying max', async () => {
+    render(
+      <FilterGroupContext.Provider value={filterGroupContextValue}>
+        <FiltersContext.Provider value={filterContextValue}>
+          <RangeInput />
+        </FiltersContext.Provider>
+      </FilterGroupContext.Provider>);
+    const actions = spyOnActions();
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    expect(screen.getByText('Apply')).toBeDefined();
+    userEvent.click(screen.getByText('Apply'));
+    expect(actions.setFilterOption).toHaveBeenCalledWith({
+      displayName: 'Up to 20',
+      fieldId: '123',
+      matcher: '$between',
+      selected: true,
+      value: {
+        end: {
+          matcher: '$le',
+          value: 20
+        },
+      }
+    });
+  });
+
+  it('renders correctly when clearing input', async () => {
+    render(
+      <FilterGroupContext.Provider value={filterGroupContextValue}>
+        <FiltersContext.Provider value={filterContextValue}>
+          <RangeInput />
+        </FiltersContext.Provider>
+      </FilterGroupContext.Provider>);
+    const maxTextbox = screen.getAllByRole('textbox')[1];
+    userEvent.type(maxTextbox, '20');
+    await waitFor(() => {
+      expect(maxTextbox).toHaveValue('20');
+    });
+    expect(screen.getByText('Clear min and max')).toBeDefined();
+    userEvent.click(screen.getByText('Clear min and max'));
+    expect(maxTextbox).toHaveValue('');
+  });
+});
+
+describe('Renders correctly for min and max inputs', () => {
+  beforeEach(() => {
+    mockAnswersHooks({ mockedState, mockedActions });
+  });
+  it('renders input value, clear, and apply option when inputing min and max values', async () => {
```

Comment:
This test can probably be remove since the test below `it('record proper values in state when applying min and max')` already check for all the expected text on screen AND check for the state when apply is clicked. Similarly with the describe blocks for min only and max only.